src/BuildIntegration/Microsoft.NETCore.Native.Windows.props

```diff
@@ -77,9 +77,16 @@ See the LICENSE file in the project root for more information.
       <LinkerArg Include="/OPT:ICF" />
     </ItemGroup>
 
-    <Exec Command="where /Q $(CppLinker)" IgnoreExitCode="true">
-      <Output TaskParameter="ExitCode" PropertyName="_WhereLinker"/>
+    <Exec Condition=" !Exists($(CppCompiler)) OR !Exists($(CppLinker)) " Command="getvctoolspath.bat x86_amd64" LogStandardErrorAsError="true" ContinueOnError="false" ConsoleToMSBuild="true">
+      <Output TaskParameter="ConsoleOutput" PropertyName="CppToolsDirectory"/>
     </Exec>
-    <Error Condition="'$(_WhereLinker)' != '0'" Text="Platform linker not found. Make sure to publish from a x64 Native Tools Command Prompt for VS 2017 with C++ tools installed." />
+
+    <PropertyGroup Condition=" '$(CppToolsDirectory)' != '' ">
+      <CppCompiler>$(CppToolsDirectory)cl.exe</CppCompiler>
+      <CppLinker>$(CppToolsDirectory)link.exe</CppLinker>
+      <CppLibCreator>$(CppToolsDirectory)lib.exe</CppLibCreator>
+    </PropertyGroup>
```

Comment:
@am11 Could you confirm if it does/doesn't?

Reply:
@A-And, @jkotas,

This does not work. I will need to refactor the script a bit more and call `ilc` from the batch script.
The issue is that MSBuild invokes the script as a child process by capturing the env vars of parent, but does not inject the environment variables on parent process (with or without SETLOCAL). For example, when we run this script 

I am experimenting by supplying three additional .bat files for link, lib and cl. Those will invoke vcvarsall.bat with `RuntimeIdentifier` provided by dotnet build/publish.

Meanwhile, I have one question: is there a mapping available for vcvarsall's architectures ` x86 | amd64 | x86_amd64 | x86_arm | x86_arm64 | amd64_x86 | amd64_arm | amd64_arm64` to dotnet core `RuntimeIdentifier`: `win-x64` etc.? I can define this map in targets OR use only `x86_amd64` for now. Any preferences?

Reply:
I think it would be nice to avoid wrapping the link invocation in a .bat file. It makes things hard to understand and debug when things go wrong.

There should be a very few environment variables and they have a linker command line switch equivalents. E.g. LIB and LIBPATH: https://docs.microsoft.com/en-us/cpp/build/reference/libpath-additional-libpath. Can we get the values fetched from the vc environment for these and just pass them on the command line?

If the logic to do any of this is too hard to write in msbuild or .bat file, it should be fine add  a custom  task for it to: https://github.com/dotnet/corert/tree/master/src/ILCompiler.Build.Tasks/src

> I can define this map in targets OR use only x86_amd64 for now.

Either one sounds good to me.

Reply:
> Can we get the values fetched from the vc environment for these and just pass them on the command line?

Sure that will make things neat. I will try to avoid the dependency on bat files that invoke vcvarsall.bat per invocation. I will try to compare env-vars before and after, unless there is a complex/registry-based calculation going on in the vcvarsall.bat and its invok'ees, it should be pretty doable. :)

Reply:
@jkotas, @A-And, in my testing, unless we pass `NODEFAULTLIB` linker option (which is not used at linker.exe call sites in .targets), it picks up the correct default libpath.

However, I am stuck at a false-positive: using the current master branch of dotnet/corert, I am unable to reproduce the vanilla shell error with this dogfooding nupkg approach: https://github.com/dotnet/corert/pull/5219#issuecomment-355795945. If I use myget feed, it fails as expected.

Could you please provide some pointers how to test the before/after changes reliably in local environment?

Reply:
I do not see anything wrong with how you are doing your testing. I suspect that you might have something set in the environment that makes it work, even when you do not expect it to work.